shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/rule/identifier/type/MutableDataNodeRule.java

```diff
@@ -42,9 +43,10 @@
     void remove(String tableName);
     
     /**
-     * Get data source names.
+     * Find single data node by table name.
      *
-     * @return Get data source names
+     * @param tableName table name
+     * @return single table data node
      */
-    Collection<String> getDataSourceNames();
+    Optional<DataNode> findSingleTableDataNodeByTableName(String tableName);
```

Comment:
@zjcnb Why not change this method to findSingleTableDataNode directly?

Reply:
Yes, I will adjust it.